Change game over animation to zoom instead of rotate

MagneFire opened the pull request:
The original rotation animation looked off, it did not fit the rest of the user experience.


This work succeeds https://github.com/AsteroidOS/asteroid-diamonds/pull/4.

Thanks to @beroset, @eLtMosen and @FlorentRevest for input on this change :smile:


Preview:

![aos-diamonds-go-4](https://user-images.githubusercontent.com/7857908/163689167-6ed7e79a-4af5-4d22-bbcf-efa20bc60916.gif)

MagneFire committed:
Game: Change game over animation to zoom instead of rotate.
The original rotation animation looked off, it did not fit the rest of the user experience.

Signed-off-by: Darrel Griët <dgriet@gmail.com>

beroset reviewed (approved):
Looks good to me!

FlorentRevest commented on src/Game.qml:
```diff
@@ -562,15 +561,24 @@ Item {
+            Rotation {
+                origin.x: width/2
+                origin.y: height/2
+                angle: -45
```
Instead of defining a -45 degrees rotation for the gameOver item whose parent is main whose rotation is 45 degrees, I think there should be two items, one for things that are rotated 45 degrees and one for the gameOver screen which would be "naturally" straight, without having to cancel out the rotation of its parent. wdyt?

MagneFire replied:
Fixed.

FlorentRevest commented on src/Game.qml:
```diff
@@ -1,6 +1,8 @@
 import Nemo.Configuration 1.0
 import org.nemomobile.lipstick 0.1
+import org.asteroid.utils 1.0
+import org.asteroid.controls 1.0
```
Why do these become necessary now ?

MagneFire replied:
`org.asteroid.utils` is needed to check if the display is round.
If the display is round, show a round animation of the game over overlay.
Else, it's a square overlay.

FlorentRevest commented:
I like that new animation yes :) I just had a couple of minor comments on the code.

Since we are here talking about animations I also want to bring up the grid animation at the beginning (we see it well in your gif) @eLtMosen how do you feel about this ? It's one of these things that looks more like a bug at first imo and then when you start questioning it you start thinking *maaybe* it's not a bug ahah.

MagneFire commented:
> I like that new animation yes :) I just had a couple of minor comments on the code.
> 

Thanks! I will get to them soonish :wink: 

> Since we are here talking about animations I also want to bring up the grid animation at the beginning (we see it well in your gif) @eLtMosen how do you feel about this ? It's one of these things that looks more like a bug at first imo and then when you start questioning it you start thinking _maaybe_ it's not a bug ahah.

It's definitely a bug ;)
Will fix that later. I only noticed it in the animation. Never before, so there's something else to fix.

moWerk commented:
> I like that new animation yes :) I just had a couple of minor comments on the code.
> 
> Since we are here talking about animations I also want to bring up the grid animation at the beginning (we see it well in your gif) @eLtMosen how do you feel about this ? It's one of these things that looks more like a bug at first imo and then when you start questioning it you start thinking _maaybe_ it's not a bug ahah.

Humm, i am so addicted to the game, i did not notice from looking at that animation too often now.
But i can see that being the first impression, agreed. 
Not wanting to cement the drafty boot animation "tiles rotate into place" style too much. But it would fit quite nicely here 😅

MagneFire committed:
Game: Fix animation upon app start.
Animations of tiles were enabled by default for all tiles.
This means that all tiles move (in an animated fashion) after they have been restored.
A workaround to this was already attempted by setting 'animateMove' to false before positioning them then enabling it again.
This however doesn't appear to work correctly during the restoration phase.

Change the logic so that all tiles are by default not animated when moved. Only enabled animation of the tiles when the user performs a swipe operation.

Signed-off-by: Darrel Griët <dgriet@gmail.com>

MagneFire commented:
Right, it should now be ready again for review.
Also fixed the startup grid animation issue. Now all tiles appear in place after app start.

Preview:
![aos-diamonds-go-5](https://user-images.githubusercontent.com/7857908/164312902-4ae6a5a5-bcda-4daf-a96f-da8c26d5a636.gif)

Some parts are slowed down to highlight the animation!
On `ray` there appears to be a slight lag spike upon app start, this is visible by the animation freezing for a few frames.

MagneFire commented:
The lag spike is now also visible on `narwhal` where it previously wasn't visible.
Converted to draft until I find a solution to this issue.

MagneFire commented:
Well, that's annoying, after testing it again it seems to perform perfectly fine...

Here's another preview (slowed down):
![aos-diamonds-go-6](https://user-images.githubusercontent.com/7857908/164543364-49d5ed1d-1f87-432c-adef-31d68275b52e.gif)

Preview (normal speed):
![aos-diamonds-go-7](https://user-images.githubusercontent.com/7857908/164543661-6cf13cb2-3260-4e0c-a596-8227d38af523.gif)


https://user-images.githubusercontent.com/7857908/164543854-8dcfe2eb-0c45-46e4-80ed-8eabe90dec91.mp4
